### PR TITLE
Add 'is_valid' function.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/semver.toit [hunk 1]
```diff
@@ -117,6 +119,81 @@ compare a/string b/string [--if_equal]:
   if comp != 0: return comp
   return if_equal.call
 
+is_letter_ c/int -> bool:
+  return 'A' <= c <= 'Z' or 'a' <= c <= 'z'
+
+is_digit_ c/int -> bool:
+  return '0' <= c <= '9'
+
+is_non_digit_ c/int -> bool:
```
Comment (kasperl) [inline src/semver.toit:128]: `is_valid_non_digit_`?

File: src/semver.toit [hunk 2]
```diff
@@ -117,6 +119,81 @@ compare a/string b/string [--if_equal]:
...
+  return c == '-' or is_letter_ c
+
+is_identifier_character_ c/int -> bool:
+  return is_digit_ c or is_non_digit_ c
+
+is_valid_build_ build/string -> bool:
+  parts := build.split "."
+  if parts.size == 0: return false
+  parts.do: | part/string |
+    if part.size == 0: return false
```
Comment (kasperl) [inline src/semver.toit]:
```suggestion
    if part.is_empty: return false
```

File: src/semver.toit [hunk 3]
```diff
@@ -117,6 +119,81 @@ compare a/string b/string [--if_equal]:
...
+    part.do:
+      if not is_identifier_character_ it: return false
+  return true
+
+is_valid_prerelease_ prerelease/string -> bool:
+  parts := prerelease.split "."
+  if parts.size == 0: return false
+  parts.do: | part/string |
+    if part.size == 0: return false
```
Comment (kasperl) [inline src/semver.toit]:
```suggestion
    if part.is_empty: return false
```

File: src/semver.toit [hunk 4]
```diff
@@ -117,6 +119,81 @@ compare a/string b/string [--if_equal]:
...
+  version_core := str
+
+  parts := version_core.split "."
+
+  if parts.size == 0: return false
+  if parts.size > 3: return false
+  if require_major_minor_patch and parts.size != 3: return false
+
+  parts.do: | part/string |
+    if part.size == 0: return false
```
Comment (kasperl) [inline src/semver.toit]:
```suggestion
    if part.is_empty: return false
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/semver.toit (hunk 1 above)
Comment (floitsch) [reply src/semver.toit:128]: These names come from the semver specification.
I prefer to keep them the way they are.

**Comment by floitsch** [comment]
Also git-added test.

**Pull request merged**